lib/chef/knife/core/subcommand_loader.rb

```diff
@@ -23,7 +23,7 @@ class Knife
     class SubcommandLoader
 
       MATCHES_CHEF_GEM = %r{/chef-[\d]+\.[\d]+\.[\d]+}.freeze
-      MATCHES_THIS_CHEF_GEM = %r{/chef-#{Chef::VERSION}/}.freeze
+      MATCHES_THIS_CHEF_GEM = %r{/chef-#{Chef::VERSION}(-x86-mingw32)?/}.freeze
```

Comment:
Ah I see, yep that's a good approach. So "chef-12.3.0-foo-bar" would actually be invalid, but we don't care because it won't happen realistically. So it's really only about preventing "12.3.0" matching "12.3.0.rc.0" or "12.3.0.dev" etc (i.e. the realisitc options), right?

I'll update the PR later tonight...